### ISSUE TITLE
Empty string as float yields weird results

### ISSUE THREAD

**Issue by pascaldevink** [webonyx/graphql-php#125]
When using a `Float` type field, it can be passed as a string:
```
{
  location(latitude:"54.0", longitude:"5.1") {
    country
  }
}
```
just works.

However, when passing an empty string (`""`), I would expect an error would be thrown by one of the validator rules, like `GraphQL\Validator\Rules\ArgumentsOfCorrectType`. Instead, an error is thrown deep down in `GraphQL\Executor\Values::getVariableValues`, which is called in `GraphQL\Validator\RulesQueryComplexity::buildFieldArguments()`, resulting in a weird error like `Float cannot represent non numeric value: (empty string)`.

The fact that a string can be passed is obviously wrong (although I can see how that can be blamed on PHP's relation with types), but this behaviour should be catched somewhere earlier I think.
Is there a validator rule missing or broken perhaps?

**Comment by vladar** [comment]
I started fixing this, but then I realized that something is wrong. String values shouldn't be allowed as input for floats at all. There are following checks in float type parsing:

```php
    public function parseLiteral($ast)
    {
        if ($ast instanceof FloatValueNode || $ast instanceof IntValueNode) {
            return (float)$ast->value;
        }
        return null;
    }
```

So `StringValueNode` should never be parsed as float (there are also tests for this). Can you double check if `ArgumentsOfCorrectType` is actually executed in your case?

**Comment by vladar** [comment]
Or maybe you are actually passing these arguments as variables vs literals?

**Comment by vladar** [comment]
Ok, I think I know what's going on. It is caused by `QueryComplexity` rule (when it is enabled). This rule throws when it operates on invalid query, hence your error.

I added separate check to skip query complexity analysis when there are other validation rules. Which should likely fix your issue, but may be not enough in the long run.

@mcg-web please review the commit with my fix: https://github.com/webonyx/graphql-php/commit/bc4b990946cbaf1fca34b3dff0488b46d37af242. Also does it make sense for you to add `try {} catch()` block in QueryComplexity rule?

**Comment by pascaldevink** [comment]
Sorry, I pressed the wrong button 🤦‍♂️

It seems that the `ArgumentsOfCorrectType` is not executed, or at least not before the `QueryComplexity` is. The way the rules are executed (seemingly in parallel, according to the `visitInParallel` function) makes it a little more difficult to debug and follow the trail. The complexity of the code doesn't help unfortunately 😞

**Comment by mcg-web** [comment]
@vladar your commit look good to me but this also mean that any error can prevent checking complexity this could maybe lead to a security issue ? `try/catch` block  is not a need to me.

**Comment by vladar** [comment]
@mcg-web Well, in theory it could be a security issue in some edge case. But if you never execute when there are validation errors, then it should be OK.

As for `try{} catch()` - this is an alternative solution, because this rule was throwing vs returning array of errors. With `try {} catch()` it could simply add exception message to list of errors vs throwing.

Just a thought. But if you don't mind current fix - it should be enough I guess.

**Comment by vladar** [comment]
Fix is included in 0.9.10
